src/main/java/cd/go/plugin/config/yaml/PluginSettings.java

```diff
@@ -17,10 +22,20 @@
 
     static PluginSettings fromJson(String json) {
         Map<String, String> raw = JSONUtils.fromJSON(json);
-        return new PluginSettings(raw.get(PLUGIN_SETTINGS_FILE_PATTERN));
+        PluginSettings settings = new PluginSettings();
+        settings.filePattern = raw.get(PLUGIN_SETTINGS_FILE_PATTERN);
+        String defaultAutoUpdateVal = raw.get(PLUGIN_SETTINGS_DEFAULT_AUTO_UPDATE);
+        settings.defaultGitAutoUpdate = defaultAutoUpdateVal == null || defaultAutoUpdateVal.isBlank() ? null :
```

Comment:
**question:** For me to understand, if no default value is set, the behavior of GoCD does not change, is that true? The plugin does **not** define a default value in the code directly?

Reply:
Yes, if it is not set, the plugin doesn't even return an `autoUpdate` field to GoCD, which means GoCD uses its default.